crates/client/l1-messages/src/config.rs

```diff
@@ -0,0 +1,36 @@
+use std::fs::File;
+use std::path::PathBuf;
+
+pub const DEFAULT_ETHEREUM_NODE: &str = "127.0.0.1:8545";
+pub const DEFAULT_CONTRACT_ADDRESS: &str = "0x5fbdb2315678afecb367f032d93f642f64180aa3";
+
+#[derive(Clone, PartialEq, serde::Deserialize, Debug)]
+pub struct L1MessagesWorkerConfig {
+    #[serde(default = "default_http")]
+    pub http_provider: String,
+    #[serde(default = "default_contract_address")]
+    pub contract_address: String,
+}
+
+impl TryFrom<&PathBuf> for L1MessagesWorkerConfig {
```

Comment:
I'm not a big fan of using `TryFrom` for this because it includes io reading.
Nothing explicitly prohibits it in the Rust language, but having a dedicated `new_from_file` method is more explicit imo